### ISSUE TITLE
NPE inside newFangledWiFiConnectivityCheck on CONNECTIVITY_CHANGE

### ISSUE THREAD

**Issue by zagorsky** [onnela-lab/beiwe-android#39]
Here's the error: https://sentry.io/onnela-lab/android-release/issues/557883367/

```
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.net.NetworkInfo.getType()' on a null object reference
    at org.beiwe.app.networking.NetworkUtility.newFangledWiFiConnectivityCheck(NetworkUtility.java:68)
    at org.beiwe.app.networking.NetworkUtility.canUpload(NetworkUtility.java:37)
    at org.beiwe.app.networking.PostRequest.uploadAllFiles(PostRequest.java:262)
    at org.beiwe.app.BackgroundService$1.onReceive(BackgroundService.java:355)
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:1178)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6688)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x4000010 launchParam=MultiScreenLaunchParams { mDisplayId=0 mFlags=0 } bqHint=1 (has extras) } in org.beiwe.app.BackgroundService$1@ed9bc0c
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:1188)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6688)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
```

One additional, related question: does upgrading the compile target to 26 affect how we receive CONNECTIVITY_CHANGE broadcasts?

**Comment by sesterki** [comment]
Fixed in 1c8bebfb49e693dfd17c134ee9bf56af637cc78f

The answer to the additional question is no.

**Comment by zagorsky** [comment]
Fixed in 1c8bebf, app version *higher than* 2.3.3
